types/twitch-js/index.d.ts

```diff
@@ -0,0 +1,175 @@
+// Project: https://github.com/twitch-apis/twitch-js#readme
+// Definitions by: Evan Steinkerchner <https://github.com/roundaround>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+// TypeScript v2.3 is only required because of the dependency on request.
+
+import * as WebSocket from 'ws';
+import { RequestCallback, Options as ApiOptions } from 'request';
+
+export interface Logger {
+    info: (message?: any, ...optionalParams: any[]) => void;
+    warn: (message?: any, ...optionalParams: any[]) => void;
+    error: (message?: any, ...optionalParams: any[]) => void;
+}
+
+export type Listener = (...args: any[]) => void;
+
+export class EventEmitter {
+    static defaultMaxListeners: number;
+    static listenerCount(emitter: EventEmitter, type: string | number): number;
+
+    setMaxListeners(n: number): this;
+    emit(type: string | number, ...args: any[]): boolean;
+    addListener(type: string | number, listener: Listener): this;
+    on(type: string | number, listener: Listener): this;
+    once(type: string | number, listener: Listener): this;
+    removeListener(type: string | number, listener: Listener): this;
+    removeAllListeners(type?: string | number): this;
+    listeners(type: string | number): Listener[];
+    listenerCount(type: string | number): number;
+    emits(types: ReadonlyArray<string | number>, values: ReadonlyArray<any>): void;
+}
+
+export interface ClientOptions {
+    channels?: string[];
+    connection?: {
+        server?: string,
+        port?: number,
+        reconnect?: boolean,
+        maxReconnectAttempts?: number,
+        maxReconnectInterval?: number,
+        reconnectDecay?: number,
+        reconnectInterval?: number,
+        secure?: boolean,
+        timeout?: number,
+    };
+    identity?: {
+        username?: string,
+        password?: string,
+    };
+    options?: {
+        clientId?: string,
+        debug?: boolean,
+        commandTimeout?: number,
+    };
+    logger?: Logger;
+}
+
+export interface Message {
+    raw: string;
+    tags: TagsCollection;
+    prefix: string|null;
+    command: string;
+    params: string[];
+}
+
+export interface Emote {
+    code: string;
+    id: number;
+}
+
+export interface TagsCollection {
+    [key: string]: string|boolean|number|null;
+}
+
+export interface Utilities {
+    levenshtein: (s1: string, s2: string, caseSensitive: boolean) => number;
+    raffle: Raffle;
+    symbols: (line: string) => number;
+    uppercase: (line: string) => number;
+}
+
+export interface Raffle {
+    init: (channel: string) => void;
+    enter: (channel: string, username: string) => void;
+    leave: (channel: string, username: string) => boolean;
+    pick: (channel: string) => string|null;
+    reset: (channel: string) => void;
+    count: (channel: string) => number;
+    isParticipating: (channel: string, username: string) => boolean;
+}
+
+export class Client extends EventEmitter {
```

Comment:
this should be client and not Client as per https://github.com/twitch-apis/twitch-js/blob/master/docs/Examples.md#browser